src/fhirtypes/common.ts

```diff
@@ -325,6 +332,42 @@ function getRuleValue(rule: AssignmentRule | CaretValueRule): AssignmentValueTyp
   }
 }
 
+export function listUndefinedLocalCodes(
+  codeSystem: FshCodeSystem | Instance,
+  codes: string[],
+  tank: FSHTank,
+  sourceEntity: FshEntity
+): void {
+  let undefinedCodes: string[] = [];
+  applyInsertRules(codeSystem, tank);
+  if (codeSystem instanceof FshCodeSystem) {
+    undefinedCodes = codes.filter(code => {
+      return !codeSystem.rules.some(rule => rule instanceof ConceptRule && rule.code === code);
+    });
+  } else if (codeSystem instanceof Instance) {
+    const conceptRulePath = /^(concept(\[(\d+|\+|=)\])?\.)+code$/;
```

Comment:
I don't remember the order of operations, but I assume that at this point we've already normalized out the indented rules so they are in the full path format, right?  Just want to be sure, as this would not work for indented rules otherwise.